server/handlers.go

```diff
@@ -290,20 +290,22 @@ func (ah *ActivityHandler) handleCreatedActivity(msg *clientmodels.Message, subs
 			return reason
 		}
 
-		channelID, userIDs, err = ah.getChatChannelIDAndUsersID(chat)
+		var channel *model.Channel
+		channel, userIDs, err = ah.getChatChannelAndUsersID(chat)
 		if err != nil {
-			ah.plugin.GetAPI().LogWarn("Unable to get original channel id", "error", err.Error())
+			ah.plugin.GetAPI().LogWarn("Unable to resolve chat to channel", "error", err.Error())
 			return metrics.DiscardedReasonOther
 		}
+		channelID = channel.Id
 
-		senderID, _ = ah.plugin.GetStore().TeamsToMattermostUserID(msg.UserID)
-		if ah.plugin.GetSelectiveSync() {
-			if shouldSync, errSync := ah.plugin.ChannelShouldSyncCreated(channelID, senderID); errSync != nil {
-				ah.plugin.GetAPI().LogWarn("Unable to determine if channel should sync", "error", errSync.Error())
-				return metrics.DiscardedReasonOther
-			} else if !shouldSync {
-				return metrics.DiscardedReasonSelectiveSync
-			}
+		var chatShouldSync bool
+		var discardReason string
+		chatShouldSync, _, discardReason, err = ah.plugin.ChatShouldSync(channel)
+		if err != nil {
+			ah.plugin.API.LogWarn("Failed to determine if created activity should sync", "channel_id", channel.Id, "error", err.Error())
+			return discardReason
+		} else if !chatShouldSync {
+			return discardReason
```

Comment:
`sender` was related to sync'ing only connected users. If sender was remote, receiver needed to be connected. So that complication should no longer be necessary, but we will sync incoming DMs to non-connected users, which I believe is desired.

Reply:
>  but we will sync incoming DMs to non-connected users, which I believe is desired.

To clarify, with selective sync, there can be at most one local user, and thus at most one "connected user". If that user isn't connected, the DM won't get synced at all since there won't be any credentials to "fetch" the message.

Is that ok from your perspective @sbishel?